FLT/Mathlib/Topology/Algebra/RestrictedProduct.lean

```diff
@@ -267,32 +278,76 @@ def Equiv.restrictedProductPi {ι : Type*} {ℱ : Filter ι} {n : Type*} [Fintyp
     {A : n → ι → Type*}
     {C : (j : n) → (i : ι) → Set (A j i)} :
     Πʳ i, [Π j, A j i, {f | ∀ j, f j ∈ C j i}]_[ℱ] ≃ Π j, Πʳ i, [A j i, C j i]_[ℱ] where
-      toFun x j := .mk (fun i ↦ x i j) (by filter_upwards [x.2] with i h using h j)
-      invFun y := .mk (fun i j ↦ y j i) (by simp)
-      left_inv x := by ext; rfl
-      right_inv y := by ext; rfl
+  toFun x j := congrRight (fun i t ↦ t _) (by simp +contextual [Set.MapsTo]) x
```

Comment:
If you're adding those variables, why not remove them from the def of `Equiv.restrictedProductPi`?